src/main/java/seedu/address/logic/commands/TagCommand.java

```diff
@@ -131,7 +122,27 @@ private ViewCommandResult tagPerson(Model model) throws CommandException {
 
         return new ViewCommandResult(MESSAGE_MODULE_TAG_PERSON_SUCCESS, editedPerson);
     }
-    private ViewCommandResult tagUser(Model model) throws CommandException {
+
+    private ViewCommandResult tagGroups(Model model) throws CommandException {
+        Person personToEdit;
+        if (index == null) {
+            personToEdit = model.getUser();
+        } else {
+            IndexHandler indexHandler = new IndexHandler(model);
+            personToEdit = indexHandler.getPersonByIndex(index).orElseThrow(() ->
+                    new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX));
+        }
+
+        personToEdit.addGroupTags(this.groupTags);
+        model.updateObservablePersonList(Model.COMPARATOR_CONTACT_INDEX_PERSON.reversed());
+        model.updateObservablePersonList(Model.COMPARATOR_CONTACT_INDEX_PERSON);
+
+        return (personToEdit instanceof User)
```

Comment:
Ideally, ternary operators should be simple. If they contain multiple lines of code, then it’s best to use an if statement already. But not important if it works, since you are a busy man 👨